### DISCUSSION TITLE
Is it possible to mimic the pages-router ssg+isr feature with the app-router?

### DISCUSSION THREAD

**Discussion by Zerebokep** [vercel/next.js#52499] [Help] [answered]
### Summary

The company I working for has been awarded by a govt contract to build a website, the contract required a low risk attack vector and a long time maintenance + update window (at least 5 years). The attack vector was primarily focused on DDoS attacks, which was our main reason to use next js with its SSG and ISR feature, as it offloads the most stuff to the client and I have to mention that the govt contractors declared this one as 'mission critical'.

Since next.js 13 has been released and the app router is stable now, my bosses are concerned if we still could meet the contractual requirements without sacrificing one of the two mentioned with nextjs anymore.

So my Q's are:

- Is it possible to mimic the pages-router ssg+isr feature with the app-router?
- If not, will there be any LTS for the pages router? The last blog article has mentioned that the pages router will still get support, but without any timeframe. The contract explicitly mentions that the website software isn't allowed to run on legacy, obsolete or deprecated software parts, LTS would be fine though.

Please excuse my ignorance to mention you, @leerob but the company I working for are massive next js advocates since years and I'm sure other companies might be in a similar situation.

**Comment by Zerebokep** [comment]
bump

**Comment by Nedi11** [comment]
DDOS protection has no relation to the stuff you described.

**Reply by Zerebokep** [reply]
Sure? From my understanding the next js server has way more work to do with the app router & "use server", especially when it comes to fetching contents from an REST API. Feel free to correct me if I'm wrong.

**Reply by Nedi11** [reply]
Unless you are doing some heavy compute which you can offload to the client e.x video rendering (which is not optimal to be done in a serverless environment  either way and is unlikely for a gov site to need such things) it is unrelated.
The apis that you will host using nextjs and call from the client will still call back to the server. In order to ddos you the attacker can just ddos the apis that you expose to the client.

**Reply by Nedi11** [reply]
Use cloudfare to protect you from ddos attacks

**Reply by Zerebokep** [reply]
> The apis that you will host using nextjs and call from the client will still call back to the server. In order to ddos you the attacker can just ddos the apis that you expose to the client.

That's the key point, we **want** attackers target the api server, as we have the complete analyze/ban/throttle arsenal there, which works perfectly for all our clients since 5 years. But if the attacker could switch to the frontend server to initiate a massive slowloris attack (which would trigger a server to server flood) we couldn't do much. Putting all the mitigation hard/software on both services would't fit budget-wise.

> Use cloudfare to protect you from ddos attacks

Unfortunately Cloudflare doesn't help much for experienced attackers, even as Enterprise client. They mitigate too much using caching or rules which also block normal users.

**Reply by Nedi11** [reply]
Well Next is a backend framework, in your case you might jus not want to use it at all and just use vite react, or just next "client". Using getServerSideProps would couse about the same load as server components.

**Reply by Zerebokep** [reply]
As said SSG + ISR works perfectly with the pages router, I'm just wondering if we can build the same with the app-router and "use client".

**Reply by Nedi11** [reply, marked as answer]
I dont think you can exactly. 
But lets supoose isr. The only benefit is the initial page render since it is prerendered on the server. Otherwise it is the same. To mimic that you can use a server component to fetch the data and pass it to a client component as initial data. Nextjs chaches requests I think so on sequential reqs the performance would be the like isr.
